core/domain/draft_upgrade_services_test.py

```diff
@@ -0,0 +1,119 @@
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for draft upgrade services."""
+
+from core.domain import draft_upgrade_services
+from core.domain import exp_domain
+from core.domain import exp_services
+from core.tests import test_utils
+import feconf
+
+
+class DraftUpgradeUnitTests(test_utils.GenericTestBase):
+    """Test the draft upgrade services module."""
+    EXP_ID = 'exp_id'
+    USER_ID = 'user_id'
+    OTHER_CHANGE_LIST = [exp_domain.ExplorationChange({
+        'cmd': exp_domain.CMD_EDIT_EXPLORATION_PROPERTY,
+        'property_name': 'title',
+        'new_value': 'New title'
+    })]
+    EXP_MIGRATION_CHANGE_LIST = [exp_domain.ExplorationChange({
+        'cmd': exp_domain.CMD_MIGRATE_STATES_SCHEMA_TO_LATEST_VERSION,
+        'from_version': '0',
+        'to_version': '1'
+    })]
+    DRAFT_CHANGELIST = [exp_domain.ExplorationChange({
+        'cmd': 'edit_exploration_property',
+        'property_name': 'title',
+        'old_value': None,
+        'new_value': 'Updated title'})]
+
+    def setUp(self):
+        super(DraftUpgradeUnitTests, self).setUp()
+        self.save_new_valid_exploration(self.EXP_ID, self.USER_ID)
+
+    def test_try_upgrade_with_no_version_difference(self):
+        self.assertIsNone(
+            draft_upgrade_services.try_upgrading_draft_to_exp_version(
+                self.DRAFT_CHANGELIST, 1, 1, self.EXP_ID))
+
+    def test_try_upgrade_failure_due_to_unsupported_commit_type(self):
+        exp_services.update_exploration(
+            self.USER_ID, self.EXP_ID, self.OTHER_CHANGE_LIST,
+            'Changed exploration title.')
+        exploration = exp_services.get_exploration_by_id(self.EXP_ID)
+        self.assertEqual(exploration.version, 2)
+        self.assertIsNone(
+            draft_upgrade_services.try_upgrading_draft_to_exp_version(
+                self.DRAFT_CHANGELIST, 1, exploration.version, self.EXP_ID))
+
+    def test_try_upgrade_failure_due_to_unimplemented_upgrade_methods(self):
+        exp_services.update_exploration(
+            self.USER_ID, self.EXP_ID, self.EXP_MIGRATION_CHANGE_LIST,
+            'Ran Exploration Migration job.')
+        exploration = exp_services.get_exploration_by_id(self.EXP_ID)
+        self.assertEqual(exploration.version, 2)
+        self.assertIsNone(
+            draft_upgrade_services.try_upgrading_draft_to_exp_version(
+                self.DRAFT_CHANGELIST, 1, exploration.version, self.EXP_ID))
+
+    def test_try_upgrade_success(self):
+        exp_services.update_exploration(
+            self.USER_ID, self.EXP_ID, self.EXP_MIGRATION_CHANGE_LIST,
+            'Ran Exploration Migration job.')
+        exploration = exp_services.get_exploration_by_id(self.EXP_ID)
+        # pylint: disable=protected-access
```

Comment:
This needs a corresponding pylint: enable.

Reply:
Done.